### PR TITLE
beyond the UI

### PR DESCRIPTION
**Pull request by asaha-scottlogic**
Have you (please tick each box to show completion):

 - [x] Added your blog post to a single category?
 - [x] Added a brief summary for your post? Summaries should be roughly two sentences in length and give potential readers a good idea of the contents of your post.
 - [x] Checked that the build passes?
 - [x] Checked your spelling (you can use `npm install` followed by `npx mdspell "**/{FILE_NAME}.md" --en-gb -a -n -x -t` if that's your thing)
 - [x] Ensured that your author profile contains a profile image, and a brief description of yourself? (make it more interesting than just your job title!)
 - [ ] Optimised any images in your post? They should be less than 100KBytes as a general guide.

Posts are reviewed / approved by your Regional Tech Lead.

### CODE REVIEW

File: _posts/2025-09-26-beyond-the-UI.markdown [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+You quickly realize that the storage of this system is a set of APIs that will need to be called to carry data from outside organizations into your company’s secure storage. But there’s a problem: the real APIs you need to connect to aren’t ready yet. Some are still being built, others are unstable, and a few are simply inaccessible for testing. How do you test your own system’s ability to send the ingested data when the other side of the bridge doesn’t exist?
+
+## 2. Why API Testing Matters
+
+As you dig deeper, you learn that API testing is not just a technical checkbox—it’s the foundation of reliable software. Unlike UI testing, which checks what users see and click, API testing goes behind the scenes, making sure the invisible data highways work flawlessly. It’s faster, more precise, and lets you catch bugs before they ever reach the user interface.
+
+API testing means sending requests to your endpoints, checking the responses, and making sure everything works as expected—handling errors, processing data, and staying secure.
+
+But with the real APIs missing, how do you even start?
```
Comment (silasvb) [inline _posts/2025-09-26-beyond-the-UI.markdown]: This doesn't make sense to me - It presents as:

1. It's important to do API testing
2. But the APIs to test are not yet present
3. And then you move on to talk about mocking

This could easily lead the reader to think you're testing your mocks, rather than a real system, which is a fairly obvious anti-pattern.

I think the root cause of this is is that this you're jumbling up the two messages this blog is trying to tell:

1. We tested our APIs
2. Our systems communicated with other APIs, which weren't available so we mocked them

Comment (asaha-scottlogic) [reply _posts/2025-09-26-beyond-the-UI.markdown]: its a blog about beyond the ui, covering api testing and mocking, so this is the best way to put it out there I thought, if you are too caught up with what happened at Cube, then that does not solve the purpose of my blog :(

File: _posts/2025-09-26-beyond-ui.markdown [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+API testing means sending requests to your endpoints, checking the responses, and making sure everything works as expected—handling errors, processing data, and staying secure.
+
+But with the real APIs missing, how do you even start?
+
+## 3. The Art of Mocking: Creating a Simulated World
+
+This is where the story gets interesting. Your team introduces you to the concept of API mocking. Think of it as building a lifelike movie set: you create fake versions of the APIs, complete with realistic responses, so you can test your system as if the real thing were there.
+
+Mocking lets you work in parallel—front-end and back-end teams can build and test without waiting for each other. You can simulate every scenario, from perfect data to the weirdest edge cases, and even inject errors or delays to see how your system reacts. It’s faster, cheaper, and gives you total control.
```
Comment (silasvb) [inline _posts/2025-09-26-beyond-ui.markdown]: (nit) - Combination of em-dash followed immediately by a hyphenated word is a bit odd to look at.  frontend/backend would resolve this.

File: _posts/2025-09-26-beyond-ui.markdown [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+## 4. Enter WireMock: The Hero Tool
+
+To bring this simulated world to life, your team chooses WireMock—a powerful open-source tool that acts as a stand-in for real APIs. WireMock lets you define exactly how your mock APIs should behave: what requests they accept, what responses they return, and how they handle errors or slowdowns.
```
Comment (silasvb) [inline _posts/2025-09-26-beyond-ui.markdown]: The final bit of this is not correct - A mock doesn't handle errors/slowdowns. It does let you introduce them.

File: _posts/2025-09-26-beyond-ui.markdown [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+- Does it handle errors gracefully when data is incomplete?
+- Does it retry or timeout when the network is slow?
+
+WireMock becomes your safety net, letting you build and test with speed and confidence. When the real APIs finally become available, you’re ready—your system has already been battle-tested.
+
+## 6. Lessons for Every IT Enthusiast
+
+As the project wraps up, you realize you’ve learned something valuable. API testing and mocking aren’t just technical tricks—they’re essential strategies for building robust, scalable, and compliant systems. They let you move fast, catch problems early, and deliver quality software even when the pieces aren’t all in place.
+
+WireMock, in particular, stands out as a hero tool—giving you the power to simulate, test, and validate your APIs in any scenario.
```
Comment (silasvb) [inline _posts/2025-09-26-beyond-ui.markdown]: Similar to my comment in Section 2, this is mixing API testing and Wiremock messages -  I'd recommend this sentence is focused on Wiremock, e.g. _... and validate your system in any scenario_, rather than API testing.

File: _posts/2025-09-26-beyond-ui.markdown [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+With WireMock, you can:
+
+- Create stubs for every endpoint your system needs.
+- Simulate successful data ingestion, error conditions, and even network timeouts.
+- Test your system’s resilience, error handling, and data validation—without ever touching the real APIs.
+
+## 5. The Project in Action
+
+Armed with WireMock, your team sets up mock endpoints for all the external data sources. You configure WireMock to send realistic regulatory data to your ingestion APIs, just like the real sources would. You test how your system handles valid data, invalid formats, missing fields, and unexpected errors.
```
Comment (silasvb) [inline _posts/2025-09-26-beyond-ui.markdown]: This implies that Wiremock is acting as a controller, which it's not.

Also, this is inconsistent with how you describe the use of Wiremock immediately below, where  you're discussing mocking content store rather than our external data providers - Which we didn't mock using Wiremock

**Comment by asaha-scottlogic** [comment]
[like]  Asrut Saha reacted to your message:
________________________________

**Comment by asaha-scottlogic** [comment]
[like]  Asrut Saha reacted to your message:
________________________________

File: _posts/2025-09-26-beyond-the-UI.markdown [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+## Beyond the UI: A Story of Data Ingestion, API Testing, and WireMock
+
+## 1. The Challenge Begins
+
+Imagine you’ve just joined a fintech company as part of a team tasked with building a new system to ingest regulatory data from various external sources. The stakes are high: the data must be accurate, complete, and compliant with strict regulations. The project is ambitious, and the timeline is tight.
+
+You quickly realize that the storage of this system is a set of APIs that will need to be called to carry data from outside organizations into your company’s secure storage. But there’s a problem: the real APIs you need to connect to aren’t ready yet. Some are still being built, others are unstable, and a few are simply inaccessible for testing. How do you test your own system’s ability to send the ingested data when the other side of the bridge doesn’t exist?
+
+## 2. Why API Testing Matters
```
Comment (csalt-scottlogic) [inline _posts/2025-09-26-beyond-the-UI.markdown]: What is the relevance of this section to the rest of the post?

Everything else is about API mocking, but this section is about API testing.  It doesn't make any sense in context

Comment (asaha-scottlogic) [reply _posts/2025-09-26-beyond-the-UI.markdown]: this is a blog, if I am calling it beyond ui and talking about apis and api mocking, i feel a small paragraph abt what api testing is no harm, i am writing it for an audience who may be reading abt api as a first time.

**Review by csalt-scottlogic — CHANGES_REQUESTED** [review]
I'm not sure this post is ready to publish as is, or if you're aiming it at the right target audience.  In general we're trying to showcase Scott Logic's depth of tech knowledge and industry awareness.  I don't think it really works either on that level or on the "article for beginners" level - it's a pretty high-level and light touch article which doesn't go deep enough into anything to showcase our expertise, and isn't explanatory enough to be any help in learning.

I'm not convinced there is enough material here to make a good post even with editing, so I've spoken around and asked Elliott Mason to reach out to you and discuss how you could find something interesting around this area to write about, and find an interesting aspect to your use of WireMock or the other work you did on the project, that would be worth writing about and would make for an engaging read

**Comment by asaha-scottlogic** [comment]
Ok Np. Thanks.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

File: _posts/2025-09-26-beyond-ui.markdown [hunk 7]
```diff
@@ -0,0 +1,66 @@
+---
+title: Beyond the UI
+date: 2025-09-25T00:00:00Z
+categories:
+- Tech
```
Comment (ColinEberhardt) [inline _posts/2025-09-26-beyond-ui.markdown]: I think this is better placed in the Testing category

Comment (asaha-scottlogic) [reply _posts/2025-09-26-beyond-ui.markdown]: true, changed.
